sdcm/cassandra_harry_thread.py

```diff
@@ -0,0 +1,184 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2021 ScyllaDB
+
+import os
+import re
+import uuid
+import time
+import logging
+import concurrent.futures
+
+from sdcm.loader import CassandraHarryStressExporter
+from sdcm.prometheus import nemesis_metrics_obj
+from sdcm.sct_events.loaders import CassandraHarryEvent, SCYLLA_BENCH_ERROR_EVENTS_PATTERNS
+from sdcm.utils.common import FileFollowerThread, generate_random_string, convert_metric_to_ms
+from sdcm.stress_thread import format_stress_cmd_error
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class CassandraHarryStressEventsPublisher(FileFollowerThread):
+    def __init__(self, node, harry_log_filename):
+        super().__init__()
+        self.harry_log_filename = harry_log_filename
+        self.node = str(node)
+
+    def run(self):
+        while not self.stopped():
+            exists = os.path.isfile(self.harry_log_filename)
+            if not exists:
+                time.sleep(0.5)
+                continue
+
+            for line_number, line in enumerate(self.follow_file(self.harry_log_filename)):
+                if self.stopped():
+                    break
+
+                for pattern, event in SCYLLA_BENCH_ERROR_EVENTS_PATTERNS:
```

Comment:
why do you search for Scylla Bench error event patterns?
whether remove it or add error patterns that are relevant for the C-H

Reply:
Ok